docs/src/rules/no-var.md

```diff
@@ -54,6 +54,42 @@ const CONFIG = {};
 
 :::
 
+This rule additionally supports TypeScript type syntax. There are multiple ways to declare global variables in TypeScript. Only using `var` works for all cases. See this [TypeScript playground](https://www.typescriptlang.org/play/?#code/PQgEB4CcFMDNpgOwMbVAGwJYCMC8AiAEwHsBbfUYAPgFgAoew6ZdAQxlAHN1jtX1QAb3qhRGaABdQAGUkAuUAGcJkTIk4ixAN3agAauwXLV6+ptFqJCWK1SgA6mpIB3IebGjHiIyrUa6HgC+9MEMdGDcvPygtqiKivSyEvQGkPReZuHAoM5OxK6ckvS5iC4AdEnFec5lqVWl+WUZYWAlLkpFdG2NSaC4oADkA-XlqX2Dw13VTWrjQ5kRPHzoACoAFpiKXJ2Ry+ubFTtL-PuKtez0uycbZ830i1GrNx3JdFdPB73982-HH2djb6Td6nGaIOaTe7ZRTQdCwbavGFww6I2Gwc5pOhI9F3LIdOEvejYlEQolojGkrHkryU+jQAAeAAdiJApIJAkA) for reference.
```

Comment:
```suggestion
This rule additionally supports TypeScript type syntax. There are multiple ways to declare global variables in TypeScript. Only using `var` works for all cases.
```

I think we can avoid this playground link as we already have examples below and we are working on typescript syntax support in ESLint playground.

Reply:
> we are working on typescript syntax support in ESLint playground.

Neat!

Still, I think the TypeScript playground does add context. The code example alone doesn’t show why it’s ok (better even) to use `var` instead of `let` inside a `declare global` block. I’ll gladly remove it if you’re still not convinced.

Reply:
In that case, instead of the playground link, how about if we add a code block in the docs page to explain why we need `var` in TypeScript with appropriate code comments in block?

you can see [this](https://eslint.org/docs/latest/rules/no-await-in-loop) page for reference.

Reply:
I think it's fine to link to the TypeScript playground for demonstration purposes. Our own playground won't have the type checking so not as solid of an example.